05-defensive.md

```diff
@@ -418,6 +418,167 @@ This violates another important rule of programming:
 "[always initialize from data](../../rules.html#always-initialize-from-data)".
 We'll leave it as an exercise to fix `range_overlap`.
 
+### Using a testing framework
+
+#### nose
+
+In the previous section we had a single test function that does different
+assertions on input data. This makes it easier to run many tests but there are
+better ways, and Python provides them for free. The recommended practise for
+test functions is to test one and one thing only. Let us split up the function
+and give each function a descriptive name. Good function names give a better
+overview of what is being tested, and it is easier to spot the corner cases we may
+have forgot about,  that should have been tested.
+
+~~~{.python}
+def test_nonoverlapping_returns_None():
+    assert range_overlap([ (0.0, 1.0), (5.0, 6.0) ]) == None
```

Comment:
Opinions vary. Is this case I think you could reasonably argue that `== None` is correct because `range_overlap` is at liberty to return any value that is `== None` and not just the unique `None` object.

However, having said all that, most Python programmers would probably expect to see `is None` and would be surprised if `range_overlap` did in fact return an object that was `== None` but not `is None`.